### PR TITLE
chore: update wallet bindings

### PR DESCRIPTION
**Pull request by Tuditi**
## Summary

Updates @iota/wallet to rc-40

...

## Changelog

```
- Updates @iota/wallet to rc-40
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows
-   **Mobile**
    -   [ ] iOS
    -   [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by begonaalvarezd** [comment]
`rc-36`-> `rc-39` seem to be unstable for now. We have to wait on new bindings to be released to be able to keep testing

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
`rc-36`-> `rc-39` seem to be unstable for now. We have to wait on new bindings to be released to be able to keep testing

File: packages/shared/lib/core/wallet/actions/mintNativeToken.ts [hunk 1]
```diff
@@ -21,21 +21,21 @@ export async function mintNativeToken(
...
 
-        const params: MintNativeTokenParams = {
+        const params: CreateNativeTokenParams = {
             maximumSupply: Converter.decimalToHex(maximumSupply),
             circulatingSupply: Converter.decimalToHex(circulatingSupply),
             foundryMetadata: Converter.utf8ToHex(JSON.stringify(metadata)),
         }
 
-        const mintTokenTransaction = await account.mintNativeToken(params, DEFAULT_TRANSACTION_OPTIONS)
+        const CreateNativeTokenTransaction = await account.mintNativeToken(params, DEFAULT_TRANSACTION_OPTIONS)
```
Comment (begonaalvarezd) [inline packages/shared/lib/core/wallet/actions/mintNativeToken.ts:30]:
```suggestion
        const createNativeTokenTransaction = await account.mintNativeToken(params, DEFAULT_TRANSACTION_OPTIONS)
```

**Review by begonaalvarezd — COMMENTED** [review]
added a mini naming comment

**Comment by Tuditi** [comment]
@brancoder , tests are failing due to enums being imported from '@iota/wallet'. You should import enums like from the following repository: '@iota/wallet/out/types'.

The reason is that the relative import from the line below is not correct anymore:

https://github.com/iotaledger/iota-sdk/blob/e6253342cd5d4571bb7c0c3775d4396d329dfcc5/bindings/nodejs-old/lib/bindings.ts#L8

**Comment by begonaalvarezd** [comment]
Duplicate of https://github.com/iotaledger/firefly/pull/7269

**Pull request closed without merging**